source/backend/api/Services/DispositionFileService.cs

```diff
@@ -246,6 +246,45 @@ public PimsDispositionSale GetDispositionFileSale(long dispositionFileId)
             return _dispositionFileRepository.GetDispositionFileSale(dispositionFileId);
         }
 
+        public PimsDispositionSale UpdateDispositionFileSale(long dispositionFileId, long saleId, PimsDispositionSale dispositionSale)
+        {
+            _logger.LogInformation("Updating disposition file Sale with DispositionFileId: {id}", dispositionFileId);
+            _user.ThrowIfNotAuthorized(Permissions.DispositionEdit);
+
+            var dispositionFileParent = _dispositionFileRepository.GetById(dispositionFileId);
+            if (dispositionFileId != dispositionSale.DispositionFileId || dispositionSale.DispositionSaleId != saleId || dispositionFileParent is null)
+            {
+                throw new BadRequestException("Invalid dispositionFileId.");
+            }
+
+            var updatedSale = _dispositionFileRepository.UpdateDispositionFileSale(saleId, dispositionSale);
+            _dispositionFileRepository.CommitTransaction();
+
+            return updatedSale;
+        }
+
+        public PimsDispositionSale AddDispositionFileSale(long dispositionFileId, PimsDispositionSale dispositionSale)
+        {
+            _logger.LogInformation("Adding disposition file Sale to Disposition File with Id: {id}", dispositionFileId);
+            _user.ThrowIfNotAuthorized(Permissions.DispositionEdit);
+
+            var dispositionFileParent = _dispositionFileRepository.GetById(dispositionFileId);
+            if (dispositionFileId != dispositionSale.DispositionFileId || dispositionFileParent is null)
+            {
+                throw new BadRequestException("Invalid dispositionFileId.");
+            }
```

Comment:
same here. the service does not need to get Ids just to validate them

Reply:
Controllers should not do validations, and be as simple as possible.

Reply:
Updated